src/Components/Components/src/RenderTree/EventHandlerDelegateWithClosuresComparer.cs

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Collections.Concurrent;
+using System.Reflection;
+using System.Linq.Expressions;
+
+namespace Microsoft.AspNetCore.Components.RenderTree
+{
+    internal static class EventHandlerDelegateWithClosuresComparer
+    {
+        private static readonly ConcurrentDictionary<Type, Func<object, object, bool>> ClosureComparerCache = new();
+
+        internal static bool EventCallBackEquals(ref EventCallback left, ref EventCallback right)
```

Comment:
Rather than this being here, would it make sense to have it as an `Equals` method on `EventCallback`? It could still be `internal`.

The reason is that if, in the future, we decided to add more fields to `EventCallback`, it would be easier for someone to realise they need to update the equality method there than to realise this other equality method exists somewhere else.

Reply:
I thought of this, but I wasn't sure that it would qualify as the override of the Equals method for the EventCallback type in every case. It does save another specific if-case in the code inside DelegateEquals if we just override Equals. (Although technically the vtable would take over the branching).
Another option would be to add to EventCallback, but don't override Equals. That would make it more clear indeed that the equality should be adapted once another field is added to the EventCallback class.

Reply:
> Another option would be to add to EventCallback, but don't override Equals.

Agreed. Making it a `static internal` method on `EventCallback` called something like `IsEquivalentForDiffing` would suffice.